File: dart_sample/lib/demos/material/material_localizations_demo_page.dart
import 'package:material_ui/material_ui.dart';

class MaterialLocalizationsDemoPage extends StatefulWidget {
  const MaterialLocalizationsDemoPage({super.key});

  @override
  State<MaterialLocalizationsDemoPage> createState() =>
      _MaterialLocalizationsDemoPageState();
}

class _MaterialLocalizationsDemoPageState
    extends State<MaterialLocalizationsDemoPage> {
  static const List<(String, Locale)> _locales = <(String, Locale)>[
    ('English', Locale('en')),
    ('Deutsch', Locale('de')),
    ('Español', Locale('es')),
    ('Русский', Locale('ru')),
    ('中文', Locale('zh')),
  ];

  static final DateTime _sampleDate = DateTime(2015, 7, 23);

  int _selected = 0;

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 12,
      children: <Widget>[
        const Text(
          'Material localizations',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'GlobalMaterialLocalizations.delegates translates every Material '
          "string and formats dates, times and numbers with the locale's own "
          'CLDR data.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        Wrap(spacing: 8, runSpacing: 8, children: _buildLocaleButtons()),
        Localizations(
          locale: _locales[_selected].$2,
          delegates: GlobalMaterialLocalizations.delegates,
          child: Builder(builder: _buildTranslations),
        ),
      ],
    );
  }

  List<Widget> _buildLocaleButtons() {
    return <Widget>[
      for (int index = 0; index < _locales.length; index++)
        GestureDetector(
          behavior: HitTestBehavior.opaque,
          onTap: () => setState(() => _selected = index),
          child: Container(
            padding: const EdgeInsets.symmetric(horizontal: 14, vertical: 8),
            decoration: BoxDecoration(
              color: index == _selected
                  ? const Color(0xFF6750A4)
                  : const Color(0xFFEDE7F6),
              borderRadius: BorderRadius.circular(10),
            ),
            child: Text(
              _locales[index].$1,
              style: TextStyle(
                fontSize: 14,
                color: index == _selected ? Colors.white : Colors.black,
              ),
            ),
          ),
        ),
    ];
  }

  Widget _buildTranslations(BuildContext context) {
    final MaterialLocalizations localizations = MaterialLocalizations.of(
      context,
    );
    final WidgetsLocalizations widgets = WidgetsLocalizations.of(context);
    const TimeOfDay midMorning = TimeOfDay(hour: 9, minute: 32);
    const TimeOfDay evening = TimeOfDay(hour: 20, minute: 32);

    return Container(
      padding: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: const Color(0xFFEDE7F6),
        borderRadius: BorderRadius.circular(12),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        spacing: 6,
        children: <Widget>[
          _buildRow('textDirection', widgets.textDirection.name),
          _buildRow('scriptCategory', localizations.scriptCategory.name),
          _buildRow('alertDialogLabel', localizations.alertDialogLabel),
          _buildRow(
            'toolbar',
            <String>[
              localizations.cutButtonLabel,
              localizations.copyButtonLabel,
              localizations.pasteButtonLabel,
              localizations.selectAllButtonLabel,
            ].join(' · '),
          ),
          _buildRow('formatFullDate', localizations.formatFullDate(_sampleDate)),
          _buildRow(
            'formatMediumDate',
            localizations.formatMediumDate(_sampleDate),
          ),
          _buildRow(
            'formatMonthYear',
            localizations.formatMonthYear(_sampleDate),
          ),
          _buildRow(
            'formatCompactDate',
            localizations.formatCompactDate(_sampleDate),
          ),
          _buildRow('dateHelpText', localizations.dateHelpText),
          _buildRow(
            'narrowWeekdays / firstDayOfWeekIndex',
            '${localizations.narrowWeekdays.join(' ')} · '
                '${localizations.firstDayOfWeekIndex}',
          ),
          _buildRow('timeOfDayFormat', localizations.timeOfDayFormat().name),
          _buildRow(
            'formatTimeOfDay',
            '${localizations.formatTimeOfDay(midMorning)} · '
                '${localizations.formatTimeOfDay(evening)}',
          ),
          _buildRow(
            'formatDecimal',
            '${localizations.formatDecimal(123)} · '
                '${localizations.formatDecimal(10000)}',
          ),
          _buildRow(
            'selectedRowCountTitle',
            localizations.selectedRowCountTitle(2),
          ),
          _buildRow(
            'pageRowsInfoTitle',
            localizations.pageRowsInfoTitle(1, 10, 100, false),
          ),
          _buildRow(
            'tabLabel',
            localizations.tabLabel(tabIndex: 1, tabCount: 2),
          ),
          _buildRow(
            'aboutListTileTitle',
            localizations.aboutListTileTitle('Plumix'),
          ),
          _buildRow('reorderItemUp', widgets.reorderItemUp),
        ],
      ),
    );
  }

  Widget _buildRow(String label, String value) {
    return Row(
      crossAxisAlignment: CrossAxisAlignment.start,
      spacing: 12,
      children: <Widget>[
        SizedBox(
          width: 220,
          child: Text(
            label,
            style: const TextStyle(fontSize: 13, color: Colors.black54),
          ),
        ),
        Expanded(
          child: Text(
            value,
            style: const TextStyle(fontSize: 13, color: Colors.black),
          ),
        ),
      ],
    );
  }
}

File: src/Sample/Plumix.Sample/Demos/Material/MaterialLocalizationsDemoPage.cs
using System;
using System.Collections.Generic;
using Avalonia.Media;
using Plumix.Material;
using Plumix.Rendering;
using Plumix.UI;
using Plumix.Widgets;

// Dart parity source (reference): dart_sample/lib/demos/material/material_localizations_demo_page.dart
// (exact sample parity)

namespace Plumix;

public sealed class MaterialLocalizationsDemoPage : StatefulWidget
{
    public override State CreateState() => new MaterialLocalizationsDemoPageState();
}

internal sealed class MaterialLocalizationsDemoPageState : State
{
    private static readonly IReadOnlyList<(string Label, Locale Locale)> Locales =
    [
        ("English", new Locale("en")),
        ("Deutsch", new Locale("de")),
        ("Español", new Locale("es")),
        ("Русский", new Locale("ru")),
        ("中文", new Locale("zh")),
    ];

    private static readonly DateTime SampleDate = new(2015, 7, 23);

    private int _selected;

    public override Widget Build(BuildContext context)
    {
        return new Column(
            crossAxisAlignment: CrossAxisAlignment.Stretch,
            spacing: 12.0,
            children:
            [
                new Text("Material localizations", fontSize: 20.0, color: Colors.Black),
                new Text(
                    "GlobalMaterialLocalizations.Delegates translates every Material string and "
                    + "formats dates, times and numbers with the locale's own CLDR data.",
                    fontSize: 14.0,
                    color: Color.Parse("#8A000000")),
                new Wrap(spacing: 8.0, runSpacing: 8.0, children: BuildLocaleButtons()),
                new Localizations(
                    locale: Locales[_selected].Locale,
                    delegates: GlobalMaterialLocalizations.Delegates,
                    child: new Builder(BuildTranslations)),
            ]);
    }

    private IReadOnlyList<Widget> BuildLocaleButtons()
    {
        var buttons = new List<Widget>();
        for (int index = 0; index < Locales.Count; index++)
        {
            int target = index;
            bool selected = index == _selected;
            buttons.Add(
                new GestureDetector(
                    behavior: HitTestBehavior.Opaque,
                    onTap: () => SetState(() => _selected = target),
                    child: new Container(
                        padding: EdgeInsets.Symmetric(horizontal: 14.0, vertical: 8.0),
                        decoration: new BoxDecoration(
                            Color: selected ? Color.Parse("#FF6750A4") : Color.Parse("#FFEDE7F6"),
                            BorderRadius: BorderRadius.Circular(10.0)),
                        child: new Text(
                            Locales[target].Label,
                            fontSize: 14.0,
                            color: selected ? Colors.White : Colors.Black))));
        }

        return buttons;
    }

    private static Widget BuildTranslations(BuildContext context)
    {
        MaterialLocalizations localizations = MaterialLocalizations.Of(context);
        WidgetsLocalizations widgets = WidgetsLocalizations.Of(context);
        var midMorning = new TimeOfDay(9, 32);
        var evening = new TimeOfDay(20, 32);

        return new Container(
            padding: EdgeInsets.All(16.0),
            decoration: new BoxDecoration(
                Color: Color.Parse("#FFEDE7F6"),
                BorderRadius: BorderRadius.Circular(12.0)),
            child: new Column(
                crossAxisAlignment: CrossAxisAlignment.Stretch,
                spacing: 6.0,
                children:
                [
                    BuildRow("textDirection", widgets.TextDirection.ToString()),
                    BuildRow("scriptCategory", localizations.ScriptCategory.ToString()),
                    BuildRow("alertDialogLabel", localizations.AlertDialogLabel),
                    BuildRow(
                        "toolbar",
                        string.Join(
                            " · ",
                            localizations.CutButtonLabel,
                            localizations.CopyButtonLabel,
                            localizations.PasteButtonLabel,
                            localizations.SelectAllButtonLabel)),
                    BuildRow("formatFullDate", localizations.FormatFullDate(SampleDate)),
                    BuildRow("formatMediumDate", localizations.FormatMediumDate(SampleDate)),
                    BuildRow("formatMonthYear", localizations.FormatMonthYear(SampleDate)),
                    BuildRow("formatCompactDate", localizations.FormatCompactDate(SampleDate)),
                    BuildRow("dateHelpText", localizations.DateHelpText),
                    BuildRow(
                        "narrowWeekdays / firstDayOfWeekIndex",
                        $"{string.Join(' ', localizations.NarrowWeekdays)} · "
                        + $"{localizations.FirstDayOfWeekIndex}"),
                    BuildRow("timeOfDayFormat", localizations.TimeOfDayFormat().ToString()),
                    BuildRow(
                        "formatTimeOfDay",
                        $"{localizations.FormatTimeOfDay(midMorning)} · "
                        + $"{localizations.FormatTimeOfDay(evening)}"),
                    BuildRow(
                        "formatDecimal",
                        $"{localizations.FormatDecimal(123)} · {localizations.FormatDecimal(10000)}"),
                    BuildRow("selectedRowCountTitle", localizations.SelectedRowCountTitle(2)),
                    BuildRow("pageRowsInfoTitle", localizations.PageRowsInfoTitle(1, 10, 100, false)),
                    BuildRow("tabLabel", localizations.TabLabel(1, 2)),
                    BuildRow("aboutListTileTitle", localizations.AboutListTileTitle("Plumix")),
                    BuildRow("reorderItemUp", widgets.ReorderItemUp),
                ]));
    }

    private static Widget BuildRow(string label, string value)
    {
        return new Row(
            crossAxisAlignment: CrossAxisAlignment.Start,
            spacing: 12.0,
            children:
            [
                new SizedBox(
                    width: 220.0,
                    child: new Text(label, fontSize: 13.0, color: Color.Parse("#8A000000"))),
                new Expanded(child: new Text(value, fontSize: 13.0, color: Colors.Black)),
            ]);
    }
}
